### PR TITLE
Menu refactoring using popup

### LINKED ISSUE #225: Enhance menus
- Move "New accounts" to a small button on the `accounts list` screen to reach
  - New Account
  - Recover Account

- Move "Edit" to a multi-select menu at the top right and allow to reach:
  - Edit
  - Change Pin
  - Backup
  - Delete

image
image

### PR DESCRIPTION
**Pull request by Tbaut**
- closes #225

screenshots:

### CODE REVIEW

**Comment by Tbaut** [comment]
@ltfschoen if you have time could you give it a shot with an iphone to make sure things compile properly, I could only test on android.

**Comment by maciejhirsz** [comment]
I'm digging the UX. You're still working on this (as per the edit in the top comment)?

**Comment by Tbaut** [comment]
Nop I'm done. Removing the edit as it caused confusion

File: src/components/PopupMenu.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    MenuOptions,
+    MenuOption,
+    MenuTrigger,
+} from 'react-native-popup-menu';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import colors from '../colors';
+
+export default class PopupMenu extends React.PureComponent {
+
+
```
Comment (maciejhirsz) [inline src/components/PopupMenu.js]: Can remove the empty lines here.

File: src/screens/AccountList.js [hunk 2]
```diff
@@ -132,34 +155,57 @@ class AccountListView extends React.PureComponent {
...
+  onboardingText: {
+    flex: 5,
+    fontFamily: 'Roboto',
+    fontSize: 20,
+    color: colors.bg_text_sec,
   }
 });
+
+
+
```
Comment (maciejhirsz) [inline src/screens/AccountList.js]: Also empty lines here :)

**Comment by Tbaut** [comment]
:+1: comments addressed. I'd like to get a confirmation from @ltfschoen before merging

**Comment by ltfschoen** [comment]
> @ltfschoen if you have time could you give it a shot with an iphone to make sure things compile properly, I could only test on android.

Yes, I just got my hands on an iPhone so will look at it asap

File: src/components/PopupMenu.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
```
Comment (ltfschoen) [inline src/components/PopupMenu.js]: -2018

File: src/components/PopupMenu.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import colors from '../colors';
+
+export default class PopupMenu extends React.PureComponent {
+    render() {
+        const { onSelect, menuTriggerIconName, menuItems } = this.props
+        const menuTriggerIcon = <Icon name={menuTriggerIconName} size={35} color={colors.bg_text_sec} />
+        return (
+            <Menu
+                onSelect={onSelect}
+            >
```
Comment (ltfschoen) [inline src/components/PopupMenu.js]: single line

File: src/screens/AccountBackup.js [hunk 5]
```diff
@@ -80,16 +80,16 @@ class AccountBackupView extends React.PureComponent {
...
-          anyone to recover this account.
+          anyone to recover and access the funds of this account.
         </Text>
         <TouchableItem
           onPress={() => {
             Alert.alert(
-              'Use paper to store seed phrases',
-              `It's not recommended to transfer or store seed phrases digitally and unencrypted. Everyone who have the phrase is able to spend funds from this account.
+              'Write this recovery phrase on paper',
+              `It's not recommended to transfer or store a recovery phrase digitally and unencrypted. Anyone in possession of this recovery phrase is able to spend funds from this account.
```
Comment (ltfschoen) [inline src/screens/AccountBackup.js]: suggest `It is`

File: src/components/PopupMenu.js (hunk 3 above)
Comment (Tbaut) [reply src/components/PopupMenu.js]: Actually 2019 right? We need to update them all, but I did this one.

**Comment by fckt** [comment]
There is no such a menus on iOS. So I predict user's dissatisfaction if such a menu will appear in iOS app too. As an alternative and in order to reach coherence between platforms I'd recommend to use the following menu:

1 wspl_j5wdpjbrml69pfm0g

Reference: https://medium.com/@chunchuanlin/android-vs-ios-compare-20-ui-components-patterns-part-1-ad33c2418b45

**Comment by fckt** [comment]
the best idea - is avoid of using the "popup" menu at all

**Comment by fckt** [comment]
There is how popups should look like on iOS:

https://www.google.com/search?q=popup++ios&safe=off&client=safari&rls=en&tbm=isch&source=lnms&sa=X&ved=0ahUKEwiRof394ePgAhUpwMQBHVa4A30Q_AUICigB&biw=1202&bih=782&dpr=2

**Comment by Tbaut** [comment]
I'm sorry but I disagree. Also I believe that this PR is not the right place to discuss design decisions. Although this implementation is surely not perfect, it streamlines processes and makes the app easier to navigate. Things are not set in stone, we can change them later on (and discuss them in an issue).

**Comment by fckt** [comment]
> Also I believe that this PR is not the right place to discuss design decisions. 

where is the right place then?

**Comment by maciejhirsz** [comment]
Apple guidelines are just that, guidelines, it's not a crime to break them. Given that we are making a cross-platform app, not adhering to guidelines of the underlying OS is, maybe not ideal, but totally fine.

UI or UX issues that would be show stoppers here are things like:
+ Something is unclear or confusing.
+ Something takes more/unnecessary amount of steps to do producing friction.

On both of these points I think the UI changes are improvements. The PR is accepted, it will be merged as soon as we solve the (recurring) issues with `ring` crate compiling that are unrelated to this PR specifically.

**Pull request merged**